### ISSUE TITLE
Ensure consistency. 

### ISSUE THREAD

**Issue by jayunit100** [rnowling/bigpetstore-data-generator#38]
looks like in recent bigpetstore work, we found that   when sending

```  val parameters:Array[String] = Array(tmpDir.toString(), "10", "1000", "365.0","123456789")
```
As the args, we can easily get different data sets.  

can we make them 100% identical ?

**Comment by rnowling** [comment]
I think this is more likely the Spark driver. We need to verify that the 0.2.1 CLI is consistent. If it is, then we can look into the Spark driver.

**Comment by jayunit100** [comment]
this bash script is the tst i ran. i think it demonstrates a that the consistency isnt maintained from run to run.  hope this is useful ! ....
```
mkdir -p gen2
touch OUTPUT
for i in $(seq 1 10); do
 java -jar ./build/libs/bigpetstore-data-generator-0.2.1.jar gen2 1 1 1000.0 1222 2>&1 > /dev/null 
 cat gen2/transactions.txt | head -1 
 rm -rf gen2
 mkdir gen2
done
```

**Comment by rnowling** [comment]
@jayunit100, thanks for reporting this!  Note that there are six parameters, not five:

```
BigPetStore Data Generator

Usage: java -jar bps-data-generator-v0.2.java outputDir nStores nCustomers simulationLength [seed]

outputDir - (string) directory to write files
nStores - (int) number of stores to generate
nCustomers - (int) number of customers to generate
nPurchasingModels - (int) number of purchasing models to generate
simulationLength - (float) number of days to simulate
seed - (long) seed for RNG. If not given, one is randomly generated.
```
so you weren't actually setting the seed :)

However, you're right -- there is a problem:

```
[rnowling@miles bigpetstore-data-generator]$ java -jar ./build/libs/bigpetstore-data-generator-0.3-SNAPSHOT.jar data1/ 10 1000 10 365.0 1234
...
Generated 25618 transactions
...
[rnowling@miles bigpetstore-data-generator]$ java -jar ./build/libs/bigpetstore-data-generator-0.3-SNAPSHOT.jar data2/ 10 1000 10 365.0 1234
...
Generated 25243 transactions
```

I'll look into it!

**Comment by jayunit100** [comment]
Yup ! i ran w/ both 5 and 6 parameters, in both cases different # of transactions.  lets definetely add a unit test for this.  i think the normal case will be that folks will want identical transactions each time.  maybe a seed should be hardcoded.?

**Comment by rnowling** [comment]
I disagree -- I think users expect different transactions every time they run it.  However, it is good to allow users to specify the seed in case for when they want reproducibility.  I also agree with adding a unit tests.

**Comment by jayunit100** [comment]
okay. this is minor but, just another alternative to hardcoding a seed, is *requiring 6 args*.  The idea is to enforce a paradigm of "same input ==  same output"

**Comment by rnowling** [comment]
That's a decent compromise.  Let's go with making the seed a required
parameter.

**Comment by jayunit100** [comment]
w00t :)

**Comment by rnowling** [comment]
@jayunit100 can you create an integration test to help me find where the deterministic behavior breaks down?

In particular, can you run two simulations with the same seed and compare:
* Store locations
* Customer first names
* Customer last names
* Customer locations
* For transactions with the same customer id and transaction id:
    * Store Ids
    * Transaction times
    * Purchases

Look at the CLI driver for how to use the Simulation class to do this.

This will help me a lot.  Thanks!
